framework/py/flwr/common/object_store/object_store.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower abstract ObjectStore definition."""
+
+from abc import ABC
+from typing import Optional
+
+
+class ObjectStore(ABC):
+    """Abstract base class for `ObjectStore` implementations.
+
+    This class defines the interface for an object store that can store, retrieve, and
+    delete objects identified by keys.
+    """
+
+    def put(self, object_id: str, object_content: bytes) -> None:
+        """Put an object into the store.
+
+        Parameters
+        ----------
+        object_id : str
+            The object_id under which to store the object.
+        object_content : bytes
+            The serialized object to store.
+        """
+        raise NotImplementedError
```

Comment:
Since we mark `ObjectStore` as an abstract base class, it's better to mark its methods as abstract methods instead of rasing a NotImplementedError manually. Same comment for other methods.

```suggestion
    @abstractmethod
    def put(self, object_id: str, object_content: bytes) -> None:
        """Put an object into the store.

        Parameters
        ----------
        object_id : str
            The object_id under which to store the object.
        object_content : bytes
            The serialized object to store.
        """
```